### PR TITLE
Softmax remove

### PR DESCRIPTION
**Pull request by montaguegabe**
Also make loss.backward() first thing after loss

### CODE REVIEW

**Comment by ByzanTine** [comment]
Why do you need to move total_loss += loss line around?

**Comment by montaguegabe** [comment]
At the time I was worried it would interfere with the backward function. It is more clear to calculate a value and then immediately backprop it than to calculate it, have it sitting around, and then do it

**Comment by ByzanTine** [comment]
Sure, readability buys me. Just to make sure I understand correctly.
